pilot/pkg/model/credentials/resource.go

```diff
@@ -30,6 +30,9 @@ const (
 	// take the form kubernetes-gateway://namespace/name. They are pulled from the config cluster.
 	KubernetesGatewaySecretType    = "kubernetes-gateway"
 	kubernetesGatewaySecretTypeURI = KubernetesGatewaySecretType + "://"
+	// BuiltinGatewaySecretType is the name of a SDS secret that uses the workloads own mTLS certificate
+	BuiltinGatewaySecretType    = "builtin"
```

Comment:
Just a little hacky here, not a good ux

Reply:
> Understand, so this is a way to auto provide certificates for gateway, that any client can connect to it via tls/https. And it does not require client cert at all.
> 
> Sounds good

Hey guys, I just would like to throw my 2 cents here since I have some chat with @johnzheng1975. I don't think providing a self-signed certs gonna be a good solution at least for AWS, and it may cause other confusion.

the problem that I can see is the AWS ALB strips off the SNI fields after terminating the TLS. if we configure any hosts in gateway resource, the gw envoy listener will not be able to handle this TLS connection. We will end up with a single wildcard gw but use different VS and hosts fields to differentiate different domains. That may break the mode if someone wants multiple gateways.

This can be the issue only to AWS ALB since I can see Azure application gateway can somehow preserve the SNI.

Reply:
I don't see what that has to do with the certificate though. If you cannot accept that limitation then you cannot use AWS ALB -- TLS --> GW. Which is fine - but this is intended for people that _do_ want that. This is documented in https://istio.io/latest/docs/ops/common-problems/network-issues/#configuring-sni-routing-when-not-sending-sni btw.

Reply:
Sorry, I didn't know this limitation is documented there. I think that's fair